OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -81,14 +81,17 @@ public virtual void Tick(Actor self)
 			else
 				realignTick = 0;
 
-			var df = DesiredFacing ?? (facing != null ? facing.Facing : TurretFacing);
-			TurretFacing = Util.TickFacing(TurretFacing, df, info.ROT);
+			self.World.AddFrameEndTask(w =>
```

Comment:
What specifically are you trying to solve with this change? This seems like it should make the problem worse, not better, because a turret can no longer adjust facing before it fires.